kiwi_stackbuild_plugin/tasks/system_stash.py

```diff
@@ -0,0 +1,128 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with kiwi.  If not, see <http://www.gnu.org/licenses/>
+#
+"""
+usage: kiwi-ng system stash -h | --help
+       kiwi-ng system stash --root=<directory> --target-dir=<directory>
+           [--tag=<name>]
+           [--maintainer=<name>]
+           [--name=<container_name>]
+       kiwi-ng system stash help
+
+commands:
+    stash
+        create a container from the given root directory
+    stash help
+        show manual page for stash command
+
+options:
+    --root=<directory>
+        the path to the root directory, usually the result of
+        a former system prepare or build call
+    --target-dir=<directory>
+        the target directory to store the container files
+    --tag=<name>
+        the tag name for the container.
+        By default set to: latest
+    --maintainer=<name>
+        the maintainer name used in the history/author and label
+        section of the container metadata.
+        By default set to: KIWI
+    --name=<container_name>
+        the name of the container as it appears in the registry.
+        By default set to: kiwi-stash-DATE
+"""
+import os
+import logging
+from datetime import datetime
+from typing import Dict
+
+from kiwi.tasks.base import CliTask
+from kiwi.help import Help
+from kiwi.defaults import Defaults
+from kiwi.privileges import Privileges
+from kiwi.container import ContainerImage
+from kiwi.container.setup import ContainerSetup
+from kiwi.path import Path
+
+
+log = logging.getLogger('kiwi')
+
+
+class SystemStashTask(CliTask):
+    def process(self) -> None:
+        self.manual = Help()
+        if self.command_args.get('help') is True:
+            return self.manual.show('kiwi::system::stash')
+
+        Privileges.check_for_root_permissions()
+
+        abs_target_dir_path = os.path.abspath(
+            self.command_args['--target-dir']
+        )
+        Path.create(abs_target_dir_path)
+
+        container_name = self.command_args['--name'] or 'kiwi-stash-{0}'.format(
+            datetime.utcnow().strftime('%Y%m%dt%H%M%S')
+        )
+        container_file = ''.join(
+            [
+                abs_target_dir_path, '/', container_name,
+                '.' + Defaults.get_platform_name(),
+                '.docker.tar'
+            ]
+        )
+        container_config = SystemStashTask._get_container_config(
+            container_name,
+            self.command_args['--tag'],
+            self.command_args['--maintainer']
+        )
+
+        container_setup = ContainerSetup.new(
+            'docker', self.command_args['--root'], container_config
+        )
+        container_setup.setup()
+        container_image = ContainerImage.new(
+            'docker', self.command_args['--root'], container_config
+        )
+        self.filename = container_image.create(
+            container_file
```

Comment:
Here I would check if the root tree includes some base image inside the `<root>/image` directory and, if so, base the image on that one. This way we can manage fully stackable builds.

Reply:
I do that now by reading the XML description. If there is none or anything wrong with it the stash will fail at that point, which is ok imho as we only want to stash kiwi created trees. The author and other information is taken from the XML description and used in the stash container config. Maybe that part can be done better and you can see the container config if you look at the tests. Let me know if this looks complete from a stash container perspective. Thanks